overlore/llm/open_ai.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from typing import Any, cast
+
+from openai import OpenAI
+
+from overlore.eternum.constants import Realms, ResourceIds, Winner
+from overlore.eternum.types import ResourceAmounts, Villager
+from overlore.llm.constants import (
+    AGENT_TEMPLATE,
+    BELLIGERENT,
+    EVENTS_EXTENSION,
+    HAPPINESS,
+    HUNGER,
+    PREVIOUS_TOWNHALL_EXTENSION,
+    ROLE,
+    SEX,
+    WORLD_SYSTEM_TEMPLATE,
+)
+from overlore.sqlite.constants import EventType
+from overlore.sqlite.types import StoredEvent
+from overlore.utils import get_enum_name_by_value, str_to_json
+
+
+class OpenAIHandler:
```

Comment:
if agreed on we can create an issue for the backlog:

the 'nl' stuff can be refactored so we have a class that just handles the text formatting logic that can be used across different LLM services

Reply:
Yeah completely agree on this

Reply:
issue'ed:https://github.com/The-Overlore/lore-machine/issues/35